Support bypassing by partitions (#13581)
Alluxio/alluxio: docs/en/core-services/Catalog.md

    thrift://metastore_host:9083 default
```

> If you want to specify a config file for the UDB, please append an option `-o catalog.db.config.file` to
> `attachdb` command.

Reviewer: nits in language tuning:

```suggestion
To specify a configuration file for the UDB, append an option `-o catalog.db.config.file` to
`attachdb` command.
```

> After each time you made changes to the config file, you can use `alluxio table sync` to apply the changes.

Reviewer: ```suggestion
Each time the configuration file is changed, you can use `alluxio table sync` to apply the changes.
```

The configuration file is in JSON format, and can contain these configurations:

> 1. Tables and partitions bypassing specification:

Reviewer: please explain here what do we mean by "Tables and partitions bypassing "

Author: Added in 9ad0139.

    ```json
    {
      "bypass": {
        "tables": [
          "table1",

>           {"table": "table2", "partitions": ["table2_part1", "table2_part2"]}

Reviewer 2: (nit) Different from the number of tables in the database, there can be tens of thousands of partitions in single tables. Can we also come up with an easy way for user to configure the bypass partitions, like bypass major part of partitions? We can do it in a follow-up pr.

Author: @HelloHorizon 
> come up with an easy way for user to configure the bypass partitions, like bypass major part of partitions

Yes, that's the plan.

@maobaolong Baolong, could you please shed some light on how we should handle partition specifications to make it practical for real use cases?

Reviewer 3: With the bypass table and partition feature, we can use it to bypass some older tables or partitions and mount a small range of tables or partitions to alluxio.

It just like a rolling forward window which stand for a range of active valuable dataset(partitions), the window can move forward by a time unit, let's say it is a week, so we just want alluxio cache the current week partition, and bypass the older partitions, if some query need the older partition, it will read data from under filesystem directly rather than read it from alluxio.

Author: @maobaolong thanks for you input. I will come up with an implementation outline and discuss with you later. Resolve this for now.

    thrift://metastore_host:9083 default
```

#### UDB Configuration File

To specify a configuration file for the UDB, append an option `-o catalog.db.config.file` to
`attachdb` command.
Each time the configuration file is changed, you can use `alluxio table sync` to apply the changes.

The configuration file is in JSON format, and can contain these configurations:

1. Tables and partitions bypassing specification:

    You can specify some tables and partitions to be bypassed from Alluxio, so that they will not be
    cached in Alluxio, instead clients will be directed to access them directly from the UDB. 
    This can be helpful when some tables and partitions are large, and accommodating them in the cache
    is undesirable.

    ```json
    {
      "bypass": {
        "tables": [
          "table1",
          {"table": "table2", "partitions": ["table2_part1", "table2_part2"]}
        ]
      }
    }
    ```

    You can specify which tables and partitions within these tables should be bypassed from Alluxio.
    By specifying only the table name, all partitions of that table, if any, will be bypassed. Otherwise, 

>     you can specify specific partitions to bypass.

Reviewer: can we explainthe above json file? e.g., are we by-passing entire table of `"table1"` and two partitions in `table2`? Am I correct? better to relate to our examples.

Author: Yes, you are correct. Clarified in 6664afc4.
